cfn-resources/cluster/cmd/resource/mappings.go

```diff
@@ -646,3 +646,51 @@ func setClusterRequest(currentModel *Model) (*admin.AdvancedClusterDescription,
 	clusterRequest.TerminationProtectionEnabled = currentModel.TerminationProtectionEnabled
 	return clusterRequest, nil
 }
+
+func AddReplicationSpecIds(src, dest []admin.ReplicationSpec) *[]admin.ReplicationSpec {
+	zoneToID := map[string]string{}
+	providerRegionToID := map[string]string{}
+	usedIds := map[string]bool{}
+
+	for _, spec := range src {
+		specID := spec.GetId()
+		if specID == "" {
+			continue
+		}
+		if zoneName := spec.GetZoneName(); zoneName != "" {
```

Comment:
nice, so you're trying to match first by zone name and then by region? does it make sense to fail the update if we're not sure, e.g. we didn't find a matching zone or region?  or that just would mean that it's a new spec and it's ok to go?

Reply:
Exactly, first by zone name and then by region 👍 
Exactly x 2 😄 , if there is no match, that means it is probably a new replication spec